src/plugins/vis_type_timelion/server/lib/reduce.js

```diff
@@ -48,16 +48,19 @@ async function pairwiseReduce(left, right, fn) {
   });
 
   // pairwise reduce seriesLists
-  const pairwiseSeriesList = { type: 'seriesList', list: [] };
-  left.list.forEach(async (leftSeries) => {
-    const first = { type: 'seriesList', list: [leftSeries] };
-    const second = { type: 'seriesList', list: [indexedList[leftSeries[pairwiseField]]] };
-    const reducedSeriesList = await reduce([first, second], fn);
-    const reducedSeries = reducedSeriesList.list[0];
-    reducedSeries.label = leftSeries[pairwiseField];
-    pairwiseSeriesList.list.push(reducedSeries);
-  });
-  return pairwiseSeriesList;
+  return {
+    type: 'seriesList',
+    list: await Promise.all(
+      left.list.map(async (leftSeries) => {
+        const first = { type: 'seriesList', list: [leftSeries] };
+        const second = { type: 'seriesList', list: [indexedList[leftSeries[pairwiseField]]] };
+        const reducedSeriesList = await reduce([first, second], fn);
+        const reducedSeries = reducedSeriesList.list[0];
+        reducedSeries.label = leftSeries[pairwiseField];
+        return reducedSeries;
+      })
+    ),
```

Comment:
In the original implementation the order of the values in `pairwiseSeriesList.list` would be in the order which they completed the `reduce()` operation. I suspect this was unintended and the new implementation is simpler with less mutation, but please let me know if that was intentional.